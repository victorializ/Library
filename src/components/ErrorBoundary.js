import React from 'react';
import { ErrorMessage } from './UITable/ErrorMessage';

class ErrorBoundary extends React.Component { 
    constructor(props) {
        super(props);
        this.state = {
            error: '',
        };
    }
    static getDerivedStateFromError(error) {
        return { error };
    }

    render() {
        if (this.state.error) {
            return <ErrorMessage text={this.state.error.toString()} /> 
        }
        return this.props.children;
    }
}
export { ErrorBoundary };