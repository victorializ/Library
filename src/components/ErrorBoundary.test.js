import React from 'react';
import { shallow, mount } from 'enzyme';

import { ErrorBoundary } from './ErrorBoundary';

describe('ErrorBoundary component', () => {
    it('renders correctly', () => {
        expect(shallow(<ErrorBoundary />)).toMatchSnapshot();
    });
});