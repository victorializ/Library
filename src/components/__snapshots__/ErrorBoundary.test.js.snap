// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component renders correctly 1`] = `""`;
